### PR TITLE
Add an option for single-threaded mode

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: lib/rubysl/timeout.rb [hunk 1]
```diff
@@ -135,26 +135,37 @@ def self.add_timeout(time, exc)
...
-
-    begin
-      yield sec
-    ensure
-      req.abort
+
+  if Truffle::Boot.single_threaded?
+  
+    def timeout(sec, exception=Error)
+      Truffle::Debug.log_warning "threads are disabled, so timeout is being ignored"
```
Comment (eregon) [inline lib/rubysl/timeout.rb]: This should still `yield`

File: test/truffle/integration/single_threaded.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+jt ruby -Xsingle_threaded -e 'require "timeout"; Timeout.timeout(1) { sleep 2 }'
+
+# Creating objects that use finalization should work
+
+jt ruby -Xsingle_threaded -e 'Rubinius::FFI::MemoryPointer.new(1024)'
+
+# Finalizations should actually be run
+
+jt ruby -Xsingle_threaded -e 'x = Object.new; ObjectSpace.define_finalizer x, -> { exit! 0 }; x = nil; GC.start; sleep 1; y = Object.new; ObjectSpace.define_finalizer y, -> { exit! 1 }; exit! 1'
```
Comment (eregon) [inline test/truffle/integration/single_threaded.sh]: Can you move this to a `.rb` file?
It's pretty hard to read like this.

File: lib/rubysl/timeout.rb (hunk 1 above)
Comment (chrisseaton) [reply lib/rubysl/timeout.rb]: Done.

File: test/truffle/integration/single_threaded.sh (hunk 2 above)
Comment (chrisseaton) [reply test/truffle/integration/single_threaded.sh]: Done.

**Pull request merged**